workflow/controller/operator_test.go

```diff
@@ -1159,3 +1159,50 @@ func TestResourceWithOwnerReferenceTemplate(t *testing.T) {
 		assert.Equal(t, "resource-with-ownerreference-template", objectMetas["resource-cm-3"].OwnerReferences[1].Name)
 	}
 }
+
+
+var withParamAsJsonList = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: expand-with-items
+spec:
+  entrypoint: expand-with-items
+  arguments:
+    parameters:
+    - name: input
+      value: '[1,2,3,4]'
```

Comment:
How is this test testing list of lists? I don't think your unit test is actually testing the feature.

Reply:
This fix is mainly if WithParam has a json array. The controller needs to execute a step for each item in the array. If WithParam has a list of lists,  Controller will execute the referred step for main array items.
Eg: withParam: [[1,2],[3,4],[5,6]]
the step will execute for 3 times with each item and pass sub-array as argument

Reply:
Yea, but your `withParam` is not a list of lists. It's: `[1,2,3,4]`, which is a single level.